libs/cli/src/main/java/org/elasticsearch/cli/internal/JsonPrintWriter.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cli.internal;
+
+import co.elastic.logging.EcsJsonSerializer;
+
+import org.elasticsearch.core.SuppressForbidden;
+
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.time.Clock;
+
+/**
+ * {@link PrintWriter} formatting lines as JSON using a subset of fields written by {@link EcsJsonSerializer}.
+ * Anything that looks like a JSON object is immediately passed through and written as is.
+ */
+public class JsonPrintWriter extends PrintWriter {
+    private static final int INITIAL_BUFFER_SIZE = 256;
+    private static final int MAX_BUFFER_SIZE = 4 * INITIAL_BUFFER_SIZE;
+
+    private final StringBuilder buffer = new StringBuilder(INITIAL_BUFFER_SIZE);
+    private final Clock clock;
+
+    public JsonPrintWriter(OutputStream out, boolean autoFlush) {
+        this(out, autoFlush, Clock.systemUTC());
+    }
+
+    @SuppressForbidden(reason = "Override PrintWriter to emit Json")
+    protected JsonPrintWriter(OutputStream out, boolean autoFlush, Clock clock) {
+        super(out, autoFlush);
+        this.clock = clock;
+    }
+
+    @Override
+    public void write(String msg) {
+        if (isJsonObject(msg)) {
+            super.write(msg);
+        } else {
+            synchronized (buffer) {
+                buffer.append(msg);
+            }
+        }
+    }
+
+    @Override
+    public void println(String msg) {
+        final String json;
+        if (isJsonObject(msg)) {
+            // Forward as is, ignore current buffer.
+            json = msg;
+        } else {
+            // Don't mind if another thread might miss a none-empty buffer.
+            // It's highly unlikely this is meant to be concatenated into the same line.
+            json = toJson(buffer.isEmpty() ? msg : getAndClearBuffer(msg), clock.millis());
+        }
+        super.write(json);
+        super.println();
+    }
+
+    @Override
+    public void println() {
+        flush();
+        super.println();
+    }
+
+    @Override
+    public void flush() {
+        synchronized (buffer) {
+            if (buffer.isEmpty() == false) {
+                super.write(toJson(getAndClearBuffer(), clock.millis()));
+            }
+        }
+        super.flush();
+    }
+
+    @Override
+    public void close() {
+        flush();
+        super.close();
+    }
+
+    /**
+     * Converts {@code msg} to JSON using the {@link EcsJsonSerializer}.
+     */
+    protected static String toJson(String msg, long timeMillis) {
+        StringBuilder builder = new StringBuilder(INITIAL_BUFFER_SIZE);
+        EcsJsonSerializer.serializeObjectStart(builder, timeMillis);
```

Comment:
Can we avoid adding a dependency on ecs logging? We can still output using the appropriate json keys, but could we serialize to json without any external dependency?

Reply:
Sure, I've removed it. It looked like a decent compromise, but fair point.
Though, are you fine copying over JsonUtils to properly quote strings? That looks fairly robust and I'd rather not reinvent things unless we're fine to just escape the very basic things such as `"`, `\`, `\n`, `\r`, `\b`, ...

Reply:
Writing a util method for escaping strings sounds good. I'm not sure how complex it needs to be though. This is specifically for messages from the JDK. I don't expect eg non-printable characters that JsonUtils tries to handle.

Reply:
Ok, in that case I'll use something much simpler that just escapes as little as absolutely necessary 👍